src/Formula.elm

```diff
@@ -713,6 +893,18 @@ spaces =
     ignore zeroOrMore (\char -> char == ' ')
 
 
+{-| String representation of a Substitution
+-}
+strSubstitution : Substitution -> String
+strSubstitution s =
+    "("
+        ++ Dict.foldl
+            (\key -> \value -> \meh -> key ++ "->" ++ strTerm value ++ ", ")
```

Comment:
Tu ignorujes/zahadzujes predch. vysledok foldu, takze to nepospajas, ale vysledoku bude iba posledna "premenna->term,".
Plus to dava `,` aj za poslednym, co tak (samozrejme asi so zatvorkami naokolo)
```elm
s
  |> Dict.toList
  |> List.map (\(v,t) -> v ++ "->" ++ strTerm t)
  |> String.join ","
```

A pridaj si unit testy (s viac premennymi v substitucii ;)

Reply:
Oki